packages/engine/Source/Core/PolygonGeometryLibrary.js

```diff
@@ -466,12 +469,294 @@ PolygonGeometryLibrary.polygonOutlinesFromHierarchy = function (
   return polygons;
 };
 
+const scratchRhumbIntersection = new Cartographic();
+function computeEquatorIntersectionRhumb(start, end, ellipsoid) {
+  const c0 = ellipsoid.cartesianToCartographic(start, scratchCartographic0);
+  const c1 = ellipsoid.cartesianToCartographic(end, scratchCartographic1);
+
+  if (Math.sign(c0.latitude) === Math.sign(c1.latitude)) {
+    return;
+  }
+
+  scratchRhumbLine.setEndPoints(c0, c1);
+
+  const intersection = scratchRhumbLine.findIntersectionWithLatitude(
+    0,
+    scratchRhumbIntersection
+  );
+
+  if (!defined(intersection)) {
+    return;
+  }
+
+  let minLongitude = Math.min(c0.longitude, c1.longitude);
+  let maxLongitude = Math.max(c0.longitude, c1.longitude);
+
+  if (Math.abs(maxLongitude - minLongitude) > CesiumMath.PI) {
+    // Crosses IDL, flip min and max
+    const swap = minLongitude;
+    minLongitude = maxLongitude;
+    maxLongitude = swap;
+  }
+
+  if (
+    intersection.longitude < minLongitude ||
+    intersection.longitude > maxLongitude
+  ) {
+    return;
```

Comment:
I'm trying to understand why the intersection might be outside the longitude range of the line. Does this only happen if the line spanned more than 180 degrees of longitude? Does `EllipsoidRhumbLine.prototype.findIntersectionWithLatitude` already handle the IDL correctly?

Reply:
So how we currently handle intersections with `EllipspoidRhumbLines`, we use the entire rhumb line circle or spiral. It is not bound by the start and end points used to created the rhumb line. This check ensures the intersection returned _is_ bound by those end points.